frontend/src/MuiForms/CustomTextInput.tsx

```diff
@@ -12,40 +14,48 @@ interface EntitySelectorProps {
   InputProps?: any
 }
 
-export default function EntitySelector(props: EntitySelectorProps) {
+export default function CustomTextInput(props: CustomTextInputProps) {
   const { onChange, value: currentValue, label, formContext } = props
 
-  const _onChange = (e) => {
-    onChange(e.target.value)
-  }
-
   const theme = useTheme()
 
+  const _onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value)
+  }
+
   return (
-    <TextField
-      {...props}
-      size='small'
-      sx={{
-        input: {
-          color: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-        label: {
-          WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-        '& .MuiInputBase-input.Mui-disabled': {
-          WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-      }}
-      onChange={_onChange}
-      variant={!formContext.editMode ? 'standard' : 'outlined'}
-      label={label}
-      required={!formContext.editMode ? false : true}
-      value={currentValue || ''}
-      disabled={!formContext.editMode}
-      InputProps={{
-        ...props.InputProps,
-        disableUnderline: !formContext.editMode ? true : false,
-      }}
-    />
+    <Fragment key={label}>
+      <Typography sx={{ fontWeight: 'bold' }}>{label}</Typography>
+      <TextField
+        size='small'
+        sx={{
+          input: {
+            color: theme.palette.mode === 'light' ? 'black' : 'white',
+          },
+          label: {
+            WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
+          },
+          '& .MuiInputBase-input.Mui-disabled': {
+            WebkitTextFillColor: currentValue
+              ? theme.palette.mode === 'light'
+                ? 'black'
+                : 'white'
+              : theme.palette.mode === 'light'
+              ? '#535353'
+              : '#c8c8c8',
+          },
+          fontStyle: currentValue ? 'unset' : 'italic',
+        }}
+        onChange={_onChange}
+        variant={!formContext.editMode ? 'standard' : 'outlined'}
+        required={!formContext.editMode ? false : true}
```

Comment:
Hard to tell what `editMode` is as `formContext` is typed as `any`. But assuming this is a boolean you can just do `required={formContext.editMode}` or if it's not a boolean you can do `required={!!formContext.editMode}`